docling_core/utils/file.py

```diff
@@ -25,6 +25,33 @@ class DocumentStream(BaseModel):
     stream: BytesIO
 
 
+def resolve_remote_filename(
+    http_url: AnyHttpUrl, response_headers: Dict[str, str]
+) -> str:
+    """Resolves the filename from a remote url and its response headers.
+
+    Args:
+        source AnyHttpUrl: The source http url.
+        response_headers Dict: Headers received while fetching the remote file.
+
+    Returns:
+        str: The actual filename of the remote url.
+    """
+    fname = None
+    # try to get filename from response header
+    if cont_disp := response_headers.get("Content-Disposition"):
+        for par in cont_disp.strip().split(";"):
+            # currently only handling directive "filename" (not "*filename")
+            if (split := par.split("=")) and split[0].strip() == "filename":
+                fname = "=".join(split[1:]).strip().strip("'\"") or None
+                break
+    # otherwise, use name from URL:
+    if fname is None:
+        fname = Path(http_url.path or "").name or "file"
```

Comment:
Could make sense instead of the hard-coded `"file"` to expose it as the default value of a kwarg like `fallback`.